src/plugin_example.rs

```diff
@@ -0,0 +1,148 @@
+        Self {
+            poll_interval: Duration::from_secs(1),
+        }
+    }
+}
+// ANCHOR_END: impl_default_config
+// ANCHOR: MyPluginSource
+#[derive(Debug)]
+struct MyPluginSource {
+    random_byte: TypedMetricId<u64>,
+}
+// ANCHOR_END: MyPluginSource
+// ANCHOR: implAlumetPlugin
+impl AlumetPlugin for MyPlugin {
+    // So we define the name of the plugin.
+    fn name() -> &'static str {
+        "MyPlugin"
+    }
+
+    // We also define it's version.
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    // We use the default config by default and on initialization.
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
+        Ok(Some(serialize_config(Config::default())?))
+    }
+
+    // We also use the default config on initialization and we deserialize the config
+    // to take in count if there is a different config than the default one.
+    fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
+        let config = deserialize_config(config)?;
+        Ok(Box::new(MyPlugin {
+            config,
+            metrics: None,
+        }))
+    }
+
+    // The start function is here to register metrics, sources and output.
+    fn start(&mut self, alumet: &mut AlumetPluginStart) -> anyhow::Result<()> {
+        // ANCHOR: createPrefixedUnit
+        let my_byte_unit: PrefixedUnit = PrefixedUnit {
+            base_unit: Unit::Byte,
+            prefix: UnitPrefix::Plain,
+        };
+        // ANCHOR_END: createPrefixedUnit
+        // ANCHOR: createMetric
+        let byte_metric =
+            alumet.create_metric::<u64>("random_byte", my_byte_unit, "Byte randomly get")?;
+        self.metrics = Some(Metrics {
+            a_metric: byte_metric,
+        });
+        // ANCHOR_END: createMetric
+        // ANCHOR: source
+        // We create a source from ThePluginSource structure.
+        let initial_source = Box::new(MyPluginSource {
+            random_byte: (self.metrics.as_ref().expect("Can't read byte_metric")).a_metric,
+        });
+
+        // Then we add it to the alumet sources, adding the poll_interval value previously defined in the config.
+        alumet.add_source(
+            initial_source,
+            TriggerSpec::at_interval(self.config.poll_interval),
+        );
+        // ANCHOR_END: source
+        Ok(())
+    }
+    // The stop function is called after all the metrics, sources and output previously
+    // registered have been stopped and unregistered.
+    fn stop(&mut self) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+// ANCHOR_END: implAlumetPlugin
+// ANCHOR: implSource
+impl Source for MyPluginSource {
+    // ANCHOR: pollFunction
+    fn poll(
+        &mut self,
+        measurements: &mut MeasurementAccumulator,
+        timestamp: Timestamp,
+    ) -> Result<(), PollError> {
+        // ANCHOR: readRandom
+        let mut rng = File::open("/dev/urandom")?; // Open the "/dev/urandom" file to obtain random data
```

Comment:
- Because it's only for educational purpose, so I didn't optimized. Using the smallest possible structures as well as the least possible
- I wanted the least external dependencies

Reply:
Yes but it only works on Linux then :S

Reply:
this is a choice